www/source/blog/2018-08-18-lift-and-shift-hab.html.md

```diff
@@ -0,0 +1,396 @@
+```
+
+We also use the Habitat maintained `core/mysql` image for some deployments, let's install and export that locally also:
+
+```bash
+[STUDIO] hab pkg install core/mysql
+» Installing core/mysql
+☁ Determining latest version of core/mysql in the 'stable' channel
+☛ Verifying core/mysql/5.7.21/20180609181124
+... output curtailed
+★ Install of  core/mysql/5.7.21/20180609181124 complete with 25 new packages installed.
+```
+
+Note the package version from the installation command, we will now use this to export a Docker image locally:
+
+```
+[STUDIO] hab pkg export docker core/mysql/5.7.21/20180609181124
+```
+
+For reference, all of the plans built and maintained by the Habitat Core Team are [available here](https://github.com/habitat-sh/core-plans).
+
+The resulting artifacts are all we need for the deployment scenarios below.
+
+## Bare metal or standalone VM deployment
+
+Let's assume we want to deploy Table Setting to an Ubuntu machine and use a file-based SQLite database backend.  In a real-world setup the versioned `tar` deployment file might be stored in a dedicated artifact repository but we will assume it is locally available on the target machine here.
+
+Relying on the Habitat package naming convention, we can run the package as follows:
+
+```bash
+$ cat hab_tar_deployment_ubuntu.sh
+#!/bin/bash
+# one-time setup - packages can specify a linux user to run under, by default this is the hab user 
+sudo adduser --disabled-password --gecos "" hab
+# assumes local package file
+sudo tar zxvf *-table-setting*.tar.gz -C /
+# run app in SQLite file mode
+export DB_TYPE=sqlite_file
+# install and run with the habitat supervisor
+sudo /hab/bin/hab sup run &
+sudo /hab/bin/hab svc load habskp/table-setting
+```
+
+Running this on a locally running Ubuntu VM looks as follows:
+
+![Running on_ubuntu_vm](media/2018-08-18-lift-and-shift-hab/run_tar_ubuntu_vm.png)
+
+
+This is arguably simpler than the steps in the previous section.  We've already successfully decoupled a lot of the package internals from the deployment process.  Although some knowledge is required to run the `tar` Habitat package locally, this would be a similar set of steps for *any* Habitat package.  Now let's see what else we can do. 
+
+## Running with Docker Compose
+
+For this case, let's take a look at the Docker Compose manifest in the root of the Table Setting repository:
+
+```bash
+$ cat docker-compose.yml
+version: '3'
+services:
+    mysql:
+        image: core/mysql
+        environment:
+          HAB_MYSQL: |
+            bind = '0.0.0.0'
+            app_username = 'tablesetting'
+            app_password = 'tablesetting'
+            root_password = 'notverysecure'
+    tablesetting:
+        image: habskp/table-setting
+        ports:
+          - 8000:8000
+        environment:
+          - APP_PORT=8000
+          - DB_TYPE=mysql
+          - DB_USER=tablesetting
+          - DB_PASSWD=tablesetting
```

Comment:
Might be good to use actual environment variables here instead of hard-coding, since `docker-compose` can understand that.

On another note, it may be confusing to new Habitat users to see Habitat configuration (in your MySQL container) combined with plain environment variable configuration, which is specific to how you've written your program.